### ISSUE TITLE
Issue Deploying Wave Dashboards Source with api 52 Features

### ISSUE THREAD

**Issue by dmrx** [forcedotcom/cli#1058]
### Summary

Issue with deploying wave dashboard (.wdash) with API v52 features (example text widget with richTextContent attribute.)

### Steps To Reproduce:

 Deploy a wave dashboard (.wdash) source  with a text widget using the CLI with new @salesforce/plugin-source installed results in error.

 Deploy a wave dashboard (.wdash) source with a text widget  using the built in 'Deploy Source' action in VSCODE results in error.

Deploy the wave dashboard (.wdash) source using the latest CLI _without_ the new 'source plugin' is fine.

### Expected result

Deploy a wave dashboard (.wdash) with a text widget  with rich formatting.

### Actual result

Failure and error:

`Problem deserializing 'setterless' property 'richTextContent': get method returned null (through reference chain: sfdc.insights.connect.api.input.DashboardStateInputRepresentation["widgets"]->sfdc.insights.connect.api.input.dashboard.widgets.parameters.TextWidgetParametersInputRepresentation["content"]->sfdc.insights.connect.api.input.dashboard.widgets.parameters.DynamicDataInputRepresentation["richTextContent"])`

### Additional information

_Feel free to attach a screenshot_.

**SFDX CLI Version**(to find the version of the CLI engine run sfdx --version): 

sfdx-cli/7.106.3 darwin-x64 node-v14.15.0

**SFDX plugin Version**(to find the version of the CLI plugin run sfdx plugins --core)
@oclif/plugin-autocomplete 0.3.0 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.2 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.10.0 (core)
@oclif/plugin-update 1.4.0-3 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
@salesforce/sfdx-trust 3.6.0 (core)
alias 1.1.10 (core)
apex 0.2.2 (core)
auth 1.6.1 (core)
config 1.2.8 (core)
custom-metadata 1.0.12 (core)
data 0.4.12 (core)
generator 1.1.8 (core)
limits 1.2.1 (core)
org 1.6.6 (core)
salesforce-alm 52.0.1 (core)
schema 1.0.7 (core)
sfdx-cli 7.106.3 (core)
sfdx-git-delta 4.6.0
sfdx-mohanc-plugins 0.0.158
**source 0.1.20**
telemetry 1.2.2 (core)
templates 51.5.0 (core)
user 1.3.0 (core)

**OS and version**: MAC OSX

**Comment by WillieRuemmele** [comment]
This seems to be a bug with the new SDR library if it's working in the current CLI and failing with the beta version, and in VSCode. Thanks 👍

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9516547](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000PYVHYA4/view)

**Comment by shetzel** [comment]
@dmrx - does it work for you if you add `--apiversion 52.0` to the command and use the source plugin?  If that works it might work without it if you update the source plugin to the latest beta version since it was updated recently with API version 52.0 types.

I'm able to deploy and retrieve WaveDashboards using --sourcepath and --metadata params.

**Comment by kaiserchiese** [comment]
Hi,
I'm having the same issue but for me the it seems is the other way round and maybe connected to other i read on some people they couldn't upgrade their CLI to API 52.0 no matter what they do. I could deploy using the new source plugin but i'm completely unable to deploy WaveDashboards with richtext since the new API version 52.0 came. I also installed and uninstalled SF CLI many times, updated the plugins, changed the api version on the CLI and also adding --apiversion 52.0 to the deploy command didn't work. Also when retrieving i'm also retrieving a version like if i retrieved with api 51. 
Only way this worked was working with package.xml with api 52 written inside but even this solves us on retrieving it doesn't on deploying as our delta deploying method is not working with this. 

This is happening in both linux machine in Bamboo and also on my Windows PC.

I'm talking with SF support since 2 weeks and i don't get a solution: 
This is the extra info i provided again today for the case:

I’m still having the issue, also against my prod org which was against where I did now the tests.
Also before that I run sfdx force:config:set apiVersion=52.0 So as you can see version is everywhere as 52.0. I’m not understanding what is happening.
Deploying with this sfdx force:source:deploy -g -u ${bamboo.sfdc.deploy.username} -p ./changed-sources/force-app/main/default -l RunLocalTests –checkonly

The -u is the variable with the name of the user nothing strange. If I include -w 10 same result. This CLI was installed yesterday again.
Inside default there is only wave/Strategic_Intent_Dashboard because is the delta folder of the changes, the user I’m using is an Admin one with all the Analytics permissions sets. This validation is against my prod org which I thought it worked but it seems not. CLI version is: sfdx-cli/7.109.0 linux-x64 node-v16.3.0
Which is totally up to date
13-Jul-2021 12:30:26	---- Deploying added and modified metadata ----
13-Jul-2021 12:30:26	➜ Deploying a delta changes to the target org sfdx-deploy-prod
13-Jul-2021 12:30:26	In Progress...
13-Jul-2021 12:30:28	WARNING: apiVersion configuration overridden at "52.0"
13-Jul-2021 12:30:29	*** Deploying with SOAP ***
13-Jul-2021 12:30:30	Job ID | 0Af2p00000EkGrdCAF
13-Jul-2021 12:30:30	SOURCE PROGRESS | ████████████████████████████████████████ | 0/0 Components
13-Jul-2021 12:30:49	SOURCE PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/1 Components
13-Jul-2021 12:30:49	
13-Jul-2021 12:30:49	
13-Jul-2021 12:30:49	=== Component Failures [1]
13-Jul-2021 12:30:49	TYPE   FILE                                                                      NAME                        PROBLEM
13-Jul-2021 12:30:49	─────  
13-Jul-2021 12:30:49	Error  sdx_sourceDeploy_pkg_1626172228932/wave/Strategic_Intent_Dashboard.wdash  Strategic_Intent_Dashboard  Problem deserializing 'setterless' property 'richTextContent': get method returned null (through reference chain: sfdc.insights.connect.api.input.DashboardStateInputRepresentation["widgets"]->sfdc.insights.connect.api.input.dashboard.widgets.parameters.TextWidgetParametersInputRepresentation["content"]->sfdc.insights.connect.api.input.dashboard.widgets.parameters.DynamicDataInputRepresentation["richTextContent"])

This is using the bamboo Linux console for the CLI but also on my PC I’m not even getting the richtext tag when retrieving so I think my PC SF CLI is also not working as it should for version 52 even I’m choosing that before too.

Now with my PC and your exact command:

PS C:\Users\IglesiPi\Documents\basf-sf> sfdx force:source:deploy -p changed-sources/force-app/main/default/wave -w 10 
-u pio.iglesias@basf.com.basf-sf -c -l RunSpecifiedTests -r COMGlobalUtilityTest
WARNING: apiVersion configuration overridden at "52.0"
*** Deploying with SOAP ***
(node:23244) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
Job ID | 0Af2p00000EkHj6CAF
SOURCE PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/1 Components

=== Component Failures [1]
TYPE   FILE                                                                      NAME                        PROBLEM
───── 
Error  sdx_sourceDeploy_pkg_1626173258777/wave/Strategic_Intent_Dashboard.wdash  Strategic_Intent_Dashboard  Problem deserializing 'setterless' property 'richTextContent': get method returned null (through reference chain: sfdc.insights.connect.api.input.DashboardStateInputRepresentation["widgets"]->sfdc.insights.connect.api.input.dashboard.widgets.parameters.TextWidgetParametersInputRepresentation["content"]->sfdc.insights.connect.api.input.dashboard.widgets.parameters.DynamicDataInputRepresentation["richTextContent"])

This is on my SF CLI in my PC against prod environment. As you see exact same error happening again. I also included the exact same file where the wave dashboard is. The runspecifiedtests is just necessary because is prod.

It seems that whatever I do my CLI is deploying as if it was v51.0 and i also checked environment variables, sfdx-project.json and that all and all is in 52.0 in both my CLI in Bamboo and on my PC.
Should I use the source beta as that solves?  Because I saw you recommend just to install and then remove by the moment.

**Comment by shetzel** [comment]
@kaiserchiese - the only way to see that error message from the org is if something is not using api version 52.0.   For example, run `sfdx plugins --core` before doing the deploy/retrieve commands and report the output.  It should look similar if not the same as this:

```
@oclif/plugin-autocomplete 0.3.0 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.2 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.10.1 (core)
@oclif/plugin-update 1.4.0-3 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
@salesforce/sfdx-trust 3.6.0 (core)
alias 1.1.10 (core)
apex 0.2.2 (core)
auth 1.7.1 (core)
config 1.2.13 (core)
custom-metadata 1.0.12 (core)
data 0.5.2 (core)
generator 1.1.8 (core)
limits 1.2.1 (core)
org 1.6.7 (core)
salesforce-alm 52.1.0 (core)
schema 1.0.8 (core)
sfdx-cli 7.110.0 (core)
telemetry 1.2.2 (core)
templates 52.1.0 (core)
user 1.3.1 (core)
```
Make sure you do not have the `salesforcedx` plugin anymore.  It was an old core plugin that is no longer used.

What is the output from running `sfdx config:list`?  Does it list a value for `apiVersion`?  If so, it MUST be 52.0 to deploy/retrieve a WaveDashboard using richTextContent.

What are your environment variables?  Is one of them `SFDX_API_VERSION`?  If so, it MUST be set to 52.0.

Do you have a `sourceApiVersion` set in your sfdx-project.json file in your project?  If so, it MUST be set to 52.0.

You can also set the `SFDX_MDAPI_TEMP_DIR` environment variable to some directory in your project.  That will save the metadata files and package.xml used to deploy.  Inspect the package.xml and ensure `<version>52.0</version>` is there and it's not some other value.

Also enable debug output when you run the deploy command by appending `--dev-debug` and maybe there is something in the output that will give some clues.
